DMTN-248.tex

```diff
@@ -46,8 +46,244 @@
+
+
+\section{Large multi-resolution postage stamps}\label{sec:bigstamps}
+
+\textbf{Proposal}: Provide larger multi-resolution postage stamps in the alert packets by binning 
+the pixels with a bin size that increases towards the edges of the images, such that several 
+arcminutes can be included.
+
+The motivation for this proposal is to enable time-domain science that relies on the rapid association 
+of low-redshift transients with their wide-area host galaxies, using algorithms such as 
+``DELIGHT: Deep Learning Identification of Galaxy Hosts of Transients using Multiresolution Images" (Förster et al. 2022).
+
+The default 6$\times$6 arcsecond postage stamp size is large enough to contain the angular diameter (30 kpc) 
+of a Milky Way-like galaxy at a redshift of 0.5, which will be a fairly typical supernova host galaxy in the LSST data set.
+The fraction of all alerts that will be low-redshift transients with wide-area host galaxies will be $<$1\% \citedsp{DMTN-102}.
+
+Host association will already be included in the alert packet as part of the \texttt{DiaObject}, as described in \citeds{DMTN-151} 
+and in Table 3 of the DPDD, including potential nearby extended and low-redshift galaxies (see columns nearbyExtObj and nearbyLowzGal).
+
+It is expected that this type of host-galaxy identification could be done by accessing the template images 
+(which are not subject to the 80 hour embargo) via the image cutout service, via which large-area cutouts could be made.
+
+Although the context provided by larger postage stamps have also been shown to better identify \texttt{DiaSources} that are glints
+(rotating debris in low earth orbit that appears as a row of point sources), ways to identify and flag \texttt{DiaSources}
+as potential glints in the AP pipeline are underway, instead of leaving this identification solely to brokers.
+
+Furthermore, although the multi-resolution aspect of the proposal aims to provide a larger area without significantly
+adding to the size of the alert packet, it is currently unclear, e.g., how much extra header information is added, or whether the
+FITS would need multiple extensions.
+
+\textbf{Recommendation}: At this time, do not increase the size of postage stamps, but potentially reconsider in the future.
+
+
+\section{Packet compression}\label{sec:compression}
+
+\textbf{Proposal}: Compress alert packets to reduce their size.
+
+The application of gzip compression could further reduce the size of a full alert to 65 KB (80\%).
+This might help to avoid alert distribution bottlenecks, lower the cost for brokers' storage needs, and potentially enable 
+additional full streams in the future -- all of which could positively impact time-domain science with the LSST.
+
+However, the time and computational resources required to compress the alert packets needs to be considered.
+For example, gzip compression at 50 MB/s to compress 10000 alerts could take 10000 $\times$ 0.08 MB per alert, 50 MB/s, or 
```

Comment:
I'm confused about the numbers here.  Remember that each CCD is processed independently, so there are only 50-200 alerts per CCD that need to be compressed.  At 50 MB/s compression speed (where does that come from?) that's a sub-second overhead.

It does increase complexity for an only modest gain in storage--you might say that Avro is already a binary format, and has the ability to do internal compression.

Reply:
Agree. Rewrote to say the following instead:

The time and computational resources required to compress the alert packets needs to be considered.
For a ballpark estimate, a quick search for "how fast is gzip compression" yeilds a rate of 50 MB/s.
For 50-200 alerts per ccd, that's only 16 MB to be compressed to 13 MB, and it would take about 0.32 seconds;
not a significant fraction of the 60 second alert production latency budgent.

However, compressing the alert packets forces brokers to then decompress the alerts on arrival, which would incur further delay.
Science goals requiring very low-latency alerts distribution might be negatively impacted by compression.
In general, it seems that compressing alert packets would increase complexity for an only modest gain in storage space.

Reply:
What if we just said something like, "At this point we don't believe we need to sacrifice latency in order to achieve bandwidth savings, but we may revisit this tradeoff as we begin testing alert distribution at scale in the USDF"

Reply:
Yup, keep it simple, like it -- replaced the paragraphs with that statement verbatim.